src/neo4clj/query_builder.clj

```diff
@@ -162,13 +167,27 @@
                 (str "(" (or (:ref-id to) to) ")")
                 (dissoc rel :ref-id)))))
 
+(defn- ensure-ref-id
+  [{:keys [ref-id] :as entity}]
+  (cond
+    (nil? entity) nil
+    (string? entity) entity
+    (nil? ref-id) (assoc entity :ref-id (cypher/gen-ref-id))
+    :else entity))
+
+(defn ensure-rel-nodes-have-ref-id
+  [{:keys [from to] :as rel}]
+  (update (update rel :from ensure-ref-id) :to ensure-ref-id))
```

Comment:
I think the following is easier to read:
```clojure
(-> rel
    (update :from ensure-ref-id)
    (update :to ensure-ref-id))

Reply:
Fixed